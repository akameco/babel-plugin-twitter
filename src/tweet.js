// @flow
import client from 'cheerio-httpcli'

export default (id: string) =>
  client
    .fetchSync(`https://twitter.com/akameco/status/${id}`)
    .$('.TweetTextSize--jumbo')
    .text()
