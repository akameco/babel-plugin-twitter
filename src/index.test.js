// @flow
import path from 'path'
import rimraf from 'rimraf'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

jest.mock('./tweet')
import tweet from './tweet' // eslint-disable-line import/first,import/imports-first

const code = `
export default (v, n, c = '0') => String(v).length >= n ? '' + v : (String(c).repeat(n) + v).slice(-n)
`
// $FlowFixMe
tweet.mockImplementation(() => code)

afterAll(() => {
  rimraf.sync(path.resolve(__dirname, '..', 'node_modules/.cache/'))
})

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  tests: [
    `import leftPad from 'twitter:712799807073419264'`,
    {
      title: 'use cache',
      code: `
    import leftPad from 'twitter:712799807073419264'
    `,
    },
    {
      title: 'typo',
      code: `
    import leftPad from 'tweeeeeeettteerrr:712799807073419264'
    `,
    },
  ],
})
