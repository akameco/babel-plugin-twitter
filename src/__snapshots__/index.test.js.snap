// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. twitter 1`] = `
"
import leftPad from 'twitter:712799807073419264'

      ↓ ↓ ↓ ↓ ↓ ↓

const leftPad = (v, n, c = '0') => String(v).length >= n ? '' + v : (String(c).repeat(n) + v).slice(-n);
"
`;

exports[`typo 1`] = `
"
import leftPad from 'tweeeeeeettteerrr:712799807073419264'

      ↓ ↓ ↓ ↓ ↓ ↓

import leftPad from 'tweeeeeeettteerrr:712799807073419264';
"
`;

exports[`use cache 1`] = `
"
import leftPad from 'twitter:712799807073419264'

      ↓ ↓ ↓ ↓ ↓ ↓

const leftPad = (v, n, c = '0') => String(v).length >= n ? '' + v : (String(c).repeat(n) + v).slice(-n);
"
`;
